### ISSUE TITLE
template_cloudinit_config generates different gzip output in v0.7.1

### ISSUE THREAD

**Issue by modax** [hashicorp/terraform#8408]
Hi,

It seems that starting with terraform v0.7.1 `template_cloudinit_config` generates different gzip output (while uncompressed output is the same) in some cases. This triggers `aws_instance` recreation due to changes in `user_data`. Please note that `content = " te"` is important. It is a minimal version of what triggers this problem for me. Other content might not trigger this bug.

What comes to my mind, did you upgrade the toolchain for terraform v0.7.1 (in contrast to v0.7.0 and previous versions) as I don't see any changes in ``template_cloudinit_config` itself.
### Terraform Version

v0.7.1 (official build) - broken
v0.7.0 (official build) and previous versions - OK
### Affected Resource(s)
- template_cloudinit_config
### Terraform Configuration Files

``` hcl
data "template_cloudinit_config" "test" {
  part {
    content_type = ""
    content = " te"
  }
}

output "test" {
  value = "${data.template_cloudinit_config.test.rendered}"
}
```
### Expected Behavior

That both 0.7.0 and 0.7.1 would generate the same (gzipped) output
### Actual Behavior

0.7.0 and 0.7.1 generate different gzip output which trigger aws_instance recreation in the user_data.
### Steps to Reproduce

Run terraform 0.7.0 and 0.7.1 against the same configuration (see above) like:

``` sh
/opt/terraform/0.7.0/terraform apply -state 0.7.0 > /dev/null; terraform output -state 0.7.0 -json | jq -r .test.value | sha1sum;
/opt/terraform/0.7.1/terraform apply -state 0.7.1 > /dev/null; terraform output -state 0.7.1 -json | jq -r .test.value | sha1sum
```

I get:

```
5239ee3a0f16d190cab62ebcd8a308e128485abf  -
78a76be3cb94954b7cd48e7c43ebf8c1d44789fb  -
```

while obviously values should have been the same if output values were the same.

**Comment by modax** [comment]
I have just confirmed my theory. I rebuilt terraform v0.7.1 from source with both go 1.6.3 and go 1.7. Results were:

* terraform v0.7.1 built with `go1.6.3 linux/amd64` - gzip output is the same as in official builds of <= v0.7.0
* terraform v0.7.1 built with `go1.7 linux/amd64` - gzip output is the same as in official build of v0.7.1

So it's 100% that upgrade to go 1.7 in your official toolchain caused this problem.

I'm not sure how to approach this. Probably it is not reasonable to expect that you will downgrade toolchain to go 1.6.3 (not forever anyway). But  on the other hand `gzip = true` is default for `template_cloudinit_config` so quite a few users will eventually be affected by this as `user_data` changes force recreation of `aws_instance`.

What would be your suggestion?
